From 602cc3072e6b030dd23f97b3dac96d4f7541e095 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Thu, 29 Jan 2026 12:11:18 +0800
Subject: [PATCH 30/36] liblto_plugin symbolic linkage fixes

---
 scripts/build/cc/gcc.sh | 25 +++++++++++++++++++------
 1 file changed, 19 insertions(+), 6 deletions(-)

diff --git a/scripts/build/cc/gcc.sh b/scripts/build/cc/gcc.sh
index 67afd848..f57f1732 100644
--- a/scripts/build/cc/gcc.sh
+++ b/scripts/build/cc/gcc.sh
@@ -796,15 +796,23 @@ do_gcc_core_backend() {
     cc_gcc_multilib_housekeeping cc="${prefix}/bin/${CT_TARGET}-${CT_CC}" \
         host="${host}"
 
+    liblto_plugin="liblto_plugin"
+    if [ "${CT_SYS_OS}" = "Cygwin" ]; then
+        liblto_plugin="cyglto_plugin"
+    fi
+
     # If binutils want the LTO plugin, point them to it
     if [ -d "${CT_PREFIX_DIR}/lib/bfd-plugins" -a "${build_step}" = "gcc_host" ]; then
         local gcc_version=$(cat "${CT_SRC_DIR}/gcc/gcc/BASE-VER" )
         local plugins_dir="libexec/gcc/${CT_TARGET}/${gcc_version}"
-        file="$( ls -1 "${CT_PREFIX_DIR}/${plugins_dir}/liblto_plugin."* 2>/dev/null | \
+        file="$( ls -1 "${CT_PREFIX_DIR}/${plugins_dir}/${liblto_plugin}."* 2>/dev/null | \
             sort | head -n1 || true )"
         [ -z "${file}" ] && ext="" || ext=".${file##*.}"
-        CT_DoExecLog ALL ln -sfv "../../${plugins_dir}/liblto_plugin${ext}" \
-                "${CT_PREFIX_DIR}/lib/bfd-plugins/liblto_plugin${ext}"
+
+        pushd "${CT_PREFIX_DIR}/lib/bfd-plugins"
+        CT_DoExecLog ALL ln -sfv "../../${plugins_dir}/${liblto_plugin}${ext}" \
+                "${liblto_plugin}${ext}"
+        popd
     fi
 }
 
@@ -1379,14 +1387,19 @@ do_gcc_backend() {
     cc_gcc_multilib_housekeeping cc="${prefix}/bin/${CT_TARGET}-${CT_CC}" \
         host="${host}"
 
+    liblto_plugin="liblto_plugin"
+
     # If binutils want the LTO plugin, point them to it
     if [ -d "${CT_PREFIX_DIR}/lib/bfd-plugins" -a "${build_step}" = "gcc_host" ]; then
         local gcc_version=$(cat "${CT_SRC_DIR}/gcc/gcc/BASE-VER" )
         local plugins_dir="libexec/gcc/${CT_TARGET}/${gcc_version}"
-        file="$( ls -1 "${CT_PREFIX_DIR}/${plugins_dir}/liblto_plugin."* 2>/dev/null | \
+        file="$( ls -1 "${CT_PREFIX_DIR}/${plugins_dir}/${liblto_plugin}."* 2>/dev/null | \
             sort | head -n1 || true )"
         [ -z "${file}" ] && ext="" || ext=".${file##*.}"
-        CT_DoExecLog ALL ln -sfv "../../${plugins_dir}/liblto_plugin${ext}" \
-                "${CT_PREFIX_DIR}/lib/bfd-plugins/liblto_plugin${ext}"
+
+        pushd "${CT_PREFIX_DIR}/lib/bfd-plugins"
+        CT_DoExecLog ALL ln -sfv "../../${plugins_dir}/${liblto_plugin}${ext}" \
+                "${liblto_plugin}${ext}"
+        popd
     fi
 }
-- 
2.52.0.windows.1

