From 0329ce04abb6446ee3e8775b3129f11dfdb001e8 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Wed, 14 Jan 2026 06:27:58 +0800
Subject: [PATCH] cygwin disguise

---
 mesonbuild/mesonmain.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/mesonmain.py b/mesonbuild/mesonmain.py
index ca7a184..a10ba8c 100644
--- a/mesonbuild/mesonmain.py
+++ b/mesonbuild/mesonmain.py
@@ -279,7 +279,7 @@ def run(original_args: T.List[str], mainfile: str) -> int:
     ensure_stdout_accepts_unicode()
 
     # https://github.com/mesonbuild/meson/issues/3653
-    if sys.platform == 'cygwin' and os.environ.get('MSYSTEM', '') not in ['MSYS', '']:
+    if sys.platform == 'cygwin' and os.environ.get('MSYSTEM', '') not in ['CYGWIN', 'MSYS', '']:
         mlog.error('This python3 seems to be msys/python on MSYS2 Windows, but you are in a MinGW environment')
         mlog.error('Please install it via https://packages.msys2.org/base/mingw-w64-python')
         return 2
-- 
2.52.0.windows.1

