From f434a6ede5da721e6d0d9118063e93f56c862aa8 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Sun, 18 Jan 2026 06:18:20 +0800
Subject: [PATCH] msysize

---
 SCons/Platform/cygwin.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SCons/Platform/cygwin.py b/SCons/Platform/cygwin.py
index 2353763..56f03f3 100644
--- a/SCons/Platform/cygwin.py
+++ b/SCons/Platform/cygwin.py
@@ -43,6 +43,10 @@ if sys.platform == 'win32':
 def generate(env) -> None:
     posix.generate(env)
 
+    import os, subprocess
+    env['ENV']['PATH'] = os.getenv('PATH')
+    env['SPAWN']       = (lambda sh, esc, cmd, args, env_param:
+                          subprocess.call(['sh', '-c', ' '.join(args)]))
     env['PROGPREFIX']  = ''
     env['PROGSUFFIX']  = '.exe'
     env['SHLIBPREFIX'] = ''
-- 
2.52.0.windows.1

